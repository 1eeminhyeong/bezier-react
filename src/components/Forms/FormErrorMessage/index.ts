export { default as FormErrorMessage } from './FormErrorMessage'
export type { default as FormErrorMessageProps } from './FormErrorMessage.types'
