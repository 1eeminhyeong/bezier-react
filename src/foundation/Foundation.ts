/* Internal dependencies */
import {
  Theme,
  Transition,
  Rounding,
  Elevation,
  Border,
  Spacing,
} from './index'

export interface Foundation {
  theme: Theme
  subTheme?: Theme
  spacing: typeof Spacing
  transition: typeof Transition
  rounding: typeof Rounding
  elevation: typeof Elevation
  border: typeof Border
}
