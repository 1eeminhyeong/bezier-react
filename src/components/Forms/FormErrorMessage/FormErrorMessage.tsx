/* External dependencies */
import React, { forwardRef } from 'react'

/* Internal dependencies */
import { BaseHelperText } from 'Components/Forms/BaseHelperText'
import FormErrorMessageProps from './FormErrorMessage.types'

export const FORM_ERROR_MESSAGE_TEST_ID = 'bezier-react-form-error-message'

function FormErrorMessage({
  testId = FORM_ERROR_MESSAGE_TEST_ID,
  color = 'bgtxt-orange-normal',
  children,
  ...rest
}: FormErrorMessageProps,
forwardedRef: React.Ref<HTMLParamElement>,
) {
  return (
    <BaseHelperText
      {...rest}
      ref={forwardedRef}
      type="error"
      testId={testId}
      aria-live="polite"
      color={color}
    >
      { children }
    </BaseHelperText>
  )
}

export default forwardRef(FormErrorMessage)
