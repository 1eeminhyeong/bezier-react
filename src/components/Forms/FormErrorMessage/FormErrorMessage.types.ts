/* Internal dependencies */
import { BaseHelperTextProps } from 'Components/Forms/BaseHelperText'

export default interface FormErrorMessageProps extends Omit<BaseHelperTextProps, 'type'> {}
