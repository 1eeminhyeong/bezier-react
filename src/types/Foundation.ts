/* Internal dependencies */
import { Foundation, css } from 'Foundation'

export interface WithFoundation {
  foundation?: Foundation
}

export type InjectedInterpolation = ReturnType<typeof css>

export interface WithInterpolation {
  interpolation?: InjectedInterpolation
}
