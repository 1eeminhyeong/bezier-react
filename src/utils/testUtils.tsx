/* External dependencies */
import React, { ReactElement } from 'react'
import { render as baseRender, RenderOptions } from '@testing-library/react'

/* Internal dependencies */
import { FoundationProvider, LightFoundation } from 'Foundation'
import { ChildrenComponentProps } from 'Types/ComponentProps'

function TestProviders({ children }: ChildrenComponentProps) {
  return (
    <FoundationProvider foundation={LightFoundation}>
      { children }
    </FoundationProvider>
  )
}

export function render(
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) {
  return baseRender(ui, { wrapper: TestProviders, ...options })
}
