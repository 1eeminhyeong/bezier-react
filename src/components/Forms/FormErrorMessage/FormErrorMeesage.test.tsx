/* External dependencies */
import React from 'react'

/* Internal dependencies */
import { LightFoundation } from 'Foundation'
import { render } from 'Utils/testUtils'
import FormErrorMessage, { FORM_ERROR_MESSAGE_TEST_ID } from './FormErrorMessage'
import type FormErrorMessageProps from './FormErrorMessage.types'

describe('FormErrorMessage >', () => {
  let props: FormErrorMessageProps
  const text = 'Lorem ipsum'

  beforeEach(() => {
    props = {
      id: 'test',
      children: text,
    }
  })

  const renderFormHelperText = (otherProps?: Partial<FormErrorMessageProps>) =>
    render(<FormErrorMessage {...props} {...otherProps} />)

  it('Snapshot >', () => {
    const { getByTestId } = renderFormHelperText()
    const rendered = getByTestId(FORM_ERROR_MESSAGE_TEST_ID)

    expect(rendered).toMatchSnapshot()
  })

  it('renders text with correct style when children prop is not empty', () => {
    const { getByTestId } = renderFormHelperText()
    const rendered = getByTestId(FORM_ERROR_MESSAGE_TEST_ID)

    expect(rendered).toHaveStyle(`color: ${LightFoundation.theme['bgtxt-orange-normal']}`)
  })

  it('renders nothing when children prop is empty', () => {
    const { queryByTestId } = renderFormHelperText({ children: '' })
    const rendered = queryByTestId(FORM_ERROR_MESSAGE_TEST_ID)

    expect(rendered).toBeNull()
  })

  it('should have aria-live="polite" attribute', () => {
    const { getByTestId } = renderFormHelperText()
    const rendered = getByTestId(FORM_ERROR_MESSAGE_TEST_ID)

    expect(rendered).toHaveAttribute('aria-live', 'polite')
  })
})
