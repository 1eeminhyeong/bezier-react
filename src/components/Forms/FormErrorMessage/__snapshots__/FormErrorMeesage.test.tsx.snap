// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorMessage > Snapshot > 1`] = `
.c0 {
  font-size: 13px;
  line-height: 18px;
  margin: 0px 0px 0px 0px;
  font-style: normal;
  font-weight: normal;
  color: #F4800B;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color;
  transition-property: color;
}

.c1 {
  display: block;
  text-align: left;
}

<p
  aria-live="polite"
  class="c0 c1"
  color="bgtxt-orange-normal"
  data-testid="bezier-react-form-error-message"
  id="test"
>
  Lorem ipsum
</p>
`;
